src/main/java/org/mockito/internal/creation/instance/InstantiatorProvider2Wrapper.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2018 Mockito contributors
+ * This program is made available under the terms of the MIT License.
+ */
+package org.mockito.internal.creation.instance;
+
+import org.mockito.mock.MockCreationSettings;
+import org.mockito.plugins.InstantiatorProvider;
+import org.mockito.plugins.InstantiatorProvider2;
+
+/**
+ * Map a {@link InstantiatorProvider2} onto a {@link InstantiatorProvider}
+ */
+public class InstantiatorProvider2Wrapper implements InstantiatorProvider {
```

Comment:
I suggest to call this type ```InstantiatorProvider2Adapter``` to clearly indicate the intent. Same applies to the other adapter: ```InstantiatorProviderWrapper``` -> ```InstantiatorProviderAdapter```